SQL/New_patches/2020-01-07-publication_users_edit_perm_rel_pk.sql

```diff
@@ -0,0 +1,5 @@
+CREATE TABLE temp SELECT DISTINCT * FROM publication_users_edit_perm_rel;
+DROP TABLE publication_users_edit_perm_rel;
+ALTER TABLE temp RENAME publication_users_edit_perm_rel;
```

Comment:
I don't understand why you're going through this temp table/drop table/alter table dance. Why can't you just alter the table to add the primary key? It seems like that would be much safer.

Reply:
in most upgrades we fix that type of problem with a one time use PHP script to cleanup/run at upgrade time before sourcing the patch. Doing it this way seems dangerous because the script isn't atomic if something goes wrong.

@ridz1208 what do you think?

Reply:
Good point, atomicity is an issue. Another alternative is to surround the queries with a transaction.

Reply:
A transaction wouldn't help. All the statements in this patch are DDL and cause an implicit commit in MySQL.

Reply:
@laemtl I agree with @driusan a PHP script in the single use directory is usually the solution we use.

In this case I would make a php script that would need to be run BEFORE the SQL patch (make sure to specify this in the changelog in the notes for developers and release notes and in the script itself and on this PR... everywhere) which basically removes duplications. should be pretty easy in PHP

Reply:
@laemtl not to say that your solution wouldnt work though !!  this is just... less risky and the PHP script's changes would be tracked in the history table if the script is done properly